nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

```diff
@@ -159,3 +255,139 @@
+	playsound(src, overheat_sound, 100, 1)
+	do_sparks(4, FALSE, src)
+	weapon_state = STATE_OVERLOAD
+	alignment = 0
+	freq = 0
+	say("WARNING! Critical heat density, emergency venting and shutdown initiated!")
+	atmos_spawn_air("water_vapor=200;TEMP=1000")
+	heat = max_heat
+	charge = 0
+	return
+
+/obj/machinery/ship_weapon/energy/proc/handle_alignment() //this is the basic bad stuff that happens, don't fire when your gun is at 0 alignment, or it'll blow itself up
+	var/turf/detonation_turf = get_turf(src)
+	if(heat >= (3*(max_heat/4)))
+		freq -= rand(1,4)
+	if(alignment <= 75)
+		if(prob(50))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+	if(alignment <= 50)
+		if(prob(45))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+			playsound(src, malfunction_sound, 100, 1)
+		if(prob(5))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
+	if(alignment <= 25)
+		if(prob(25))
+			do_sparks(4, FALSE, src)
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+		if(prob(25))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
+		if(prob(50))
+			var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
+			var/obj/item/projectile/beam/laser/P = new(detonation_turf)
+			//Shooting Code:
+			P.range = 6
+			P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
+			P.fire()
+			freq -= rand(1,10)
+	alignment = max(alignment-(rand(0, 4)),0)
+
+
+	// dilithium crystal alignment minigame stolen from ds13
+/obj/machinery/ship_weapon/energy/screwdriver_act(mob/user, obj/item/tool)
+	. = ..()
+	if(maint_state == MSTATE_UNBOLTED)
+		.=TRUE
+		var/sound/thesound = pick('nsv13/sound/effects/computer/beep.ogg','nsv13/sound/effects/computer/beep2.ogg','nsv13/sound/effects/computer/beep3.ogg','nsv13/sound/effects/computer/beep4.ogg','nsv13/sound/effects/computer/beep5.ogg','nsv13/sound/effects/computer/beep6.ogg','nsv13/sound/effects/computer/beep7.ogg','nsv13/sound/effects/computer/beep8.ogg','nsv13/sound/effects/computer/beep9.ogg','nsv13/sound/effects/computer/beep10.ogg','nsv13/sound/effects/computer/beep11.ogg','nsv13/sound/effects/computer/beep12.ogg',)
+		SEND_SOUND(user, thesound)
+		var/list/options = letters
+		for(var/option in options)
+			options[option] = image(icon = 'nsv13/icons/actions/engine_actions.dmi', icon_state = "[option]")
+		var/dowhat = show_radial_menu(user,src,options)
+		if(!dowhat)
+			return
+		combo += "[dowhat]"
+		combocount ++
+		to_chat(user, "<span class='warning'>You inputted [dowhat] into the command sequence.</span>")
+		playsound(src, 'sound/machines/sm/supermatter3.ogg', 20, 1)
+		if(combocount <= 4)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/atom,screwdriver_act), user),2)   // *scream  addtimer(CALLBACK(object|null, GLOBAL_PROC_REF(type/path|procstring), arg1, arg2, ... argn), time, timertype)
```

Comment:
This looks terrible, why not use a for loop to continuously let people insert new things until they've added 5? (Or if they press "cancel" somewhere)

Reply:
Some of this is repaired ds13 code. I don't exactly know how to recreate it or modify it. I think it was originally supposed to be timed